app/assets/javascripts/models/beta/projectBoard.js

```diff
@@ -68,44 +68,51 @@ export const toggleColumn = (projectBoard, column, callback) => {
 }
 
 // Drag And drop utils
-const calculatePosition = (aboveStory, bellowStory, storyState) => {
+const calculatePositions = ({ aboveStory = {}, belowStory = {}, storyState }) => {
   const aboveStoryState = aboveStory?.state;
-  const bellowStoryState = bellowStory?.state;
-  const aboveStoryPosition = aboveStory?.position;
-  const bellowStoryPosition = bellowStory?.position;
-
-  if (!bellowStory) return Number(aboveStoryPosition) + 1;
-  if (!aboveStory) return Number(bellowStoryPosition) - 1;
-  if (aboveStoryState === storyState && bellowStoryState !== storyState) return Number(aboveStoryPosition) + 1;
-  if (bellowStoryState === storyState && aboveStoryState !== storyState) return Number(bellowStoryPosition) - 1;
-
-  return (Number(bellowStoryPosition) + Number(aboveStoryPosition)) / 2;
-};
+  const belowStoryState = belowStory?.state;
+  const aboveStoryPosition = Number(aboveStory?.position);
+  const belowStoryPosition = Number(belowStory?.position);
+  const aboveStoryNewPosition = aboveStory?.newPosition;
+  const belowStoryNewPosition = belowStory?.newPosition;
```

Comment:
how about:
```js
  const { state: aboveStoryState, newPosition: aboveStoryNewPosition } = aboveStory;
  const { state: belowStoryState, newPosition: belowStoryNewPosition } = belowStory;
  const aboveStoryPosition = Number(aboveStory.position);
  const belowStoryPosition = Number(belowStory.position);
  ```

You dont need to check with `aboveStory?.` because you have a default value for them